sdks/python/apache_beam/io/filesystem.py

```diff
@@ -269,8 +270,9 @@ class FileMetadata(object):
   """
   def __init__(self, path, size_in_bytes):
     assert isinstance(path, basestring) and path, "Path should be a string"
-    assert isinstance(size_in_bytes, int) and size_in_bytes >= 0, \
-        "Size of bytes should be greater than equal to zero"
+    assert isinstance(size_in_bytes, numbers.Number) and size_in_bytes >= 0, \
```

Comment:
Can you use `isinstance(size_in_bytes, (int, long))` instead and add a comment that Windows uses longs here?  I think using numbers.Number may be too general (e.g. floats match).  (and can you remove the `import numbers` above?)